tools/convert_datasets/a2d2.py

```diff
@@ -0,0 +1,389 @@
+    (210, 50, 115):    0,  # Zebra crossing --> Road
+    (255, 0, 128):   255,  # Obstacles / trash --> Ignore ?
+    (255, 246, 143):   5,  # Poles --> Poles
+    (150, 0, 150):     0,  # RD restricted area --> Road ?
+    (204, 255, 153): 255,  # Animals --> Ignore ?
+    (238, 162, 173):   0,  # Grid structure --> Road ???
+    (33, 44, 177):     6,  # Signal corpus --> Traffic light ?
+    (180, 50, 180):    0,  # Drivable cobblestone --> Road
+    (255, 70, 185):  255,  # Electronic traffic --> Ignore ???
+    (238, 233, 191):   0,  # Slow drive area --> Road
+    (147, 253, 194):   8,  # Nature object --> Vegetation
+    (150, 150, 200):   0,  # Parking area --> Road
+    (180, 150, 200):   1,  # Sidewalk --> Sidewalk
+    (72, 209, 204):  255,  # Ego car --> Ignore
+    (200, 125, 210):   0,  # Painted driv. instr. --> Road ???
+    (159, 121, 238):   0,  # Traffic guide obj. --> Road ???
+    (128, 0, 255):     0,  # Dashed line --> Road
+    (255, 0, 255):     0,  # RD normal street --> Road
+    (135, 206, 255):  10,  # Sky --> Sky
+    (241, 230, 255):   2,  # Buildings --> Building
+    (96, 69, 143):   255,  # Blurred area --> Ignore ?
+    (53, 46, 82):    255,  # Rain dirt --> Ignore ?
+}
+
+
+def modify_label_filename(label_filepath):
+    ''' Returns a mmsegmentation-combatible label filename.
+    '''
+    label_filepath = label_filepath.replace('_label_', '_camera_')
+    label_filepath = label_filepath.replace('.png', '_labelTrainIds.png')
+    return label_filepath
+
+
+def convert_a2d2_trainids(label_filepath, ignore_id=255):
+    '''Saves a new semantic label using the A2D2 label categories.
+
+    The new image is saved into the same directory as the original image having
+    an additional suffix.
+
+    Args:
+        label_filepath: Path to the original semantic label.
+        ignore_id: Default value for unlabeled elements.
+    '''
+    # Read label file as Numpy array (H, W, 3)
+    orig_label = Image.open(label_filepath)
+    orig_label = np.array(orig_label, dtype=np.int)
+
+    # Empty array with all elements set as 'ignore id' label
+    H, W, _ = orig_label.shape
+    mod_label = ignore_id*np.ones((H, W), dtype=np.int)
+
+    seg_colors = list(SEG_COLOR_DICT_A2D2.keys())
+    for seg_color in seg_colors:
+        # The operation produce (H,W,3) array of (i,j,k)-wise truth values
+        mask = (orig_label == seg_color)
+        # Take the product channel-wise to falsify any partial match and
+        # collapse RGB channel dimension (H,W,3) --> (H,W)
+        #   Ex: [True, False, False] --> [False]
+        mask = np.prod(mask, axis=-1)
+        mask = mask.astype(np.bool)
+        # Segment masked elements with 'trainIds' value
+        mod_label[mask] = SEG_COLOR_DICT_A2D2[seg_color]
+
+    # Save new 'trainids' semantic label
+    label_filepath = modify_label_filename(label_filepath)
+    label_img = Image.fromarray(mod_label.astype(np.uint8))
+    label_img.save(label_filepath)
```

Comment:
`mmcv.imwrite`

Reply:
Changed to `mmcv.imwrite(label_img, label_filepath)`